sdk/internal/testframework/testcontext.go

```diff
@@ -0,0 +1,37 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package testframework
+
+type TestContext interface {
+	Fail(string)
+	Log(string)
+	Name() string
+}
+
+type testContext struct {
+	fail Failer
+	log  Logger
+	name string
+}
+
+type Failer func(string)
+type Logger func(string)
+type Name func() string
+
+func NewTestContext(f Failer, l Logger, name Name) TestContext {
```

Comment:
Nit: `name` to `n`

This doesnt really matter but I'd say for consistency it's better?